kartothek/io/eager.py

```diff
@@ -280,7 +274,7 @@ def read_table(
         >>> df = read_table(store, 'dataset_uuid', 'core')
 
     """
-    if concat_partitions_on_primary_index:
+    if concat_partitions_on_primary_index is not False:
```

Comment:
why was this change necessary?

Reply:
Passing `concat_partitions_on_primary_index` as anything should also raise a warning. 
If not making the above change, passing 0 will not raise the warning